docs/language/types.rst

```diff
@@ -395,37 +395,43 @@ Mappings are declared with ``mapping(keytype => valuetype)``, for example:
 
   When assigning multiple members in a struct in a mapping, it is better to create
   a storage variable as a reference to the struct, and then assign to the reference.
-  The add() function above could have been written as:
+  The ``add()`` function above can be optimized like the following.
 
   .. code-block:: solidity
 
     function add(string name, address addr) public {
-        s[name].exists = true;
-        s[name].addr = addr;
+        // assigning to a storage variable creates a reference
+        user storage s = users[name];
+
+        s.exists = true;
+        s.addr = addr;
     }
 
-  Here the storage slot for struct is calculated twice, which includes an expensive
+  Here the storage slot for the struct is calculated only once, avoiding another expensive
   keccak256 calculation.
 
-If you access a non-existing field on a mapping, all the fields will read as zero. So, it
+If you access a non-existing field on a mapping, all the fields will read as zero. It
 is common practise to have a boolean field called ``exists``. Since mappings are not iterable,
 it is not possible to ``delete`` an entire mapping itself, but individual mapping entries can be deleted.
 
 .. note::
 
   Solidity takes the keccak 256 hash of the key and the storage slot, and simply uses that
```

Comment:
This should probably say `Solidity on Ethereum` or `Solidity on EVM`.

Reply:
Isn't that also valid for Substrate?

Reply:
Yes, you're right!

Reply:
Fixed